pytest_rerunfailures.py

```diff
@@ -250,12 +255,155 @@ def _should_not_rerun(item, report, reruns):
     )
 
 
+def is_master(config):
+    return not (hasattr(config, "workerinput") or hasattr(config, "slaveinput"))
+
+
+def pytest_configure(config):
+    try:
+        from xdist.newhooks import pytest_handlecrashitem as _
+
+        if is_master(config):
+            config.failures_db = ServerStatusDB()
+        else:
+            config.failures_db = ClientStatusDB(config.workerinput['sock_port']) 
+    except ImportError:
+        config.failures_db = StatusDB()  # no-op db
```

Comment:
According to the code of `StatusDB` it is not actually a `no-op` – why?

Reply:
If xdist is not detected, then we don't need to record failures across workers, because a test crash `os.exit(1)` will crash pytest. Therefore, this functionality is irrelevant without xdist. With xdist, we need to record failures across workers because we need to reduce the number of times to rerun a test by the number of times that test has crashed a worker.

Reply:
I think I understood the goal but `StatusDB` initializes a socket … maybe there should be an actual no-op StatusDB and a base class for the classes for client and server which encapsulates the common code for these ones. But a socket and probably some other code should not be needed in the no-op version.

Reply:
Yes, but the socket is not bound to a port. My understanding is that a socket is no-op until it is bound.